### PR TITLE
Add processing to tabulator

### PR DESCRIPTION
**Pull request by chases2**
Allows processing to be done between tab states. Actually useful processing to be implemented.

Adds a `copy-only` flag to suppress this behavior.

### CODE REVIEW

File: pkg/tabulator/tabstate.go [hunk 1]
```diff
@@ -169,8 +174,8 @@ func Update(ctx context.Context, client gcs.ConditionalClient, mets *Metrics, co
 			log.WithFields(logrus.Fields{
 				"from": fromPath.String(),
 				"to":   toPath.String(),
-			}).Info("Copying state")
-			if confirm {
+			}).Info("Calculating state")
+			if copyOnly && confirm {
```
Comment (michelle192837) [inline pkg/tabulator/tabstate.go:178]: nit: If confirm = false, think neither copy nor upload should happen?

File: pkg/tabulator/tabstate.go [hunk 2]
```diff
@@ -245,3 +260,46 @@ func TabStatePath(configPath gcs.Path, tabPrefix, dashboardName, tabName string)
 	}
 	return np, nil
 }
+
+func calculateState(ctx context.Context, client gcs.Client, log *logrus.Entry, fromPath, toPath gcs.Path, confirm bool) error {
```
Comment (michelle192837) [inline pkg/tabulator/tabstate.go:264]: nit: calculateState doesn't indicate to me that this does uploads, is there a better name to clarify that?

**Review by michelle192837 — APPROVED** [review]
/hold

Looks good, couple of nits!

File: pkg/tabulator/tabstate.go [hunk 3]
```diff
@@ -245,3 +260,46 @@ func TabStatePath(configPath gcs.Path, tabPrefix, dashboardName, tabName string)
...
+	buf, err := ioutil.ReadAll(z)
+	if err != nil {
+		return fmt.Errorf("ioutil.ReadAll: %w", err)
+	}
+	var g statepb.Grid
+	if err = proto.Unmarshal(buf, &g); err != nil {
+		return fmt.Errorf("proto.Unmarshal: %w", err)
+	}
+
+	// TODO(chases2): From summarizer: filterGrid()
```
Comment (fejta) [inline pkg/tabulator/tabstate.go:284]: IMO this PR is fine but not particularly useful. 

This just demonstrates using the proto and zip packages, which seems unlikely to cause a problem.

The more interesting piece here is to inflate the grid and then deflate it.

AKA convert it into `[]InflatedGrid` and back

https://github.com/GoogleCloudPlatform/testgrid/blob/90494ebb6318b4a135aa950c1d70d087d74a81ab/pkg/updater/inflate.go#L32-L37

One idea might be to call `InflateDropAppend` with a noop column reader:

https://github.com/GoogleCloudPlatform/testgrid/blob/90494ebb6318b4a135aa950c1d70d087d74a81ab/pkg/updater/updater.go#L549

**Review by fejta — APPROVED** [review]

File: cmd/tabulator/main.go [hunk 4]
```diff
@@ -70,6 +71,7 @@ func gatherOptions() options {
 	flag.Var(&o.persistQueue, "persist-queue", "Load previous queue state from gs://path/to/queue-state.json and regularly save to it thereafter")
 	flag.StringVar(&o.creds, "gcp-service-account", "", "/path/to/gcp/creds (use local creds if empty)")
 	flag.BoolVar(&o.confirm, "confirm", false, "Upload data if set")
+	flag.BoolVar(&o.copyOnly, "copy-only", true, "Copies grid state to tab state with no transformation")
```
Comment (fejta) [inline cmd/tabulator/main.go:74]: Please default this to off until we roll it out to prod.

File: pkg/tabulator/tabstate.go (hunk 3 above)
Comment (chases2) [reply pkg/tabulator/tabstate.go:285]: I don't think we'll need to inflate/deflate the grid in order to filter it. The summarizer doesn't.

We will need to inflate/deflate to drop columns later, to determine alerting and to drop empty columns after filtering, but I intend to do that in a later PR.

File: cmd/tabulator/main.go (hunk 4 above)
Comment (chases2) [reply cmd/tabulator/main.go]: It was defaulted "off", but it wasn't as clear I don't think. Switched the flag around

File: pkg/tabulator/tabstate.go (hunk 1 above)
Comment (chases2) [reply pkg/tabulator/tabstate.go]: Yes, that's right. There was a bug here my tests didn't catch; thanks!

File: pkg/tabulator/tabstate.go (hunk 2 above)
Comment (chases2) [reply pkg/tabulator/tabstate.go]: Chose a more vague name, since this function currently "does everything" for a particular tab.

**Pull request merged**